### ISSUE TITLE
error messages while running homer_mysql_rotate.pl

### ISSUE THREAD

**Issue by dunst0** [sipcapture/homer-api#148]
Trying to setup homer sipcapture on a FreeBSD 11.0 machine but when setting up the MySQL rotation script I've got some error messages. After searching in the wiki i didn't find any info's on the problem, but maybe you have some insight on it and point me to the place where I did a configuration error.

```
DBD::mysql::db do failed: Specified key was too long; max key length is 767 bytes at ./homer_mysql_rotate.pl line 383.
Missing argument in printf at ./homer_mysql_rotate.pl line 383.
```

**Comment by lmangani** [comment]
Hi @dunst0 what version of MySQL are you running?

**Comment by dunst0** [comment]
I'm running version 5.6.35

**Comment by dunst0** [comment]
@lmangani do I need some spezial setting for the mysql server?

**Comment by adubovikov** [comment]
do you have same script ?

https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L383

I am not sure :-(

Wbr,
Alexandr

**Comment by dunst0** [comment]
@adubovikov I just checked out the source code from the homer-ui and homer-api repositories, so should be the same, I just took the rotations.ini from the repository with these values

```
[MSG_TABLE_SIZE]
    #characters
    sip_capture_call = 1500
    isup_capture_all = 3000
    webrtc_capture_all = 3000
    rtcp_capture_all = 1500
```

I thought they should be ok for using or do I have to change them?

**Comment by adubovikov** [comment]
@dunst0 

again,  compare that you have :

Missing argument in printf at ./homer_mysql_rotate.pl line 383.

and that we have here:

https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L383

and you will see the different.

just download the last version of the perl script and check it again :)

**Comment by dunst0** [comment]
@adubovikov you are right they they weren't the same. For some reason I had not the latest version.
But the error message is still there but at line 408.

```
DBD::mysql::db do failed: Specified key was too long; max key length is 767 bytes at ./homer_mysql_rotate.pl line 408.
Missing argument in printf at ./homer_mysql_rotate.pl line 408.
Failed to execute query [Specified key was too long; max key length is 767 bytes] with error: DBD::mysql::db do failed: Specified key was too long; max key length is 767 bytes at ./homer_mysql_rotate.pl line 408.
Missing argument in printf at ./homer_mysql_rotate.pl line 408.
Failed to execute query [Specified key was too long; max key length is 767 bytes] with error: DBD::mysql::db do failed: Specified key was too long; max key length is 767 bytes at ./homer_mysql_rotate.pl line 408.
Missing argument in printf at ./homer_mysql_rotate.pl line 408.
Failed to execute query [Specified key was too long; max key length is 767 bytes] with error: 
```

**Comment by adubovikov** [comment]
IMHO you have some limitation in your mysq.
please set largest prefix:
SET @@global.innodb_large_prefix = 1;

and/or change table encoding to latin1 and don't use utf8

**Comment by dougbtv** [comment]
Also, fwiw, these two lines are missing the `(255)` constraint on the key:

https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L212
https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L211

Others have it, for example...

https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L183

and

https://github.com/sipcapture/homer-api/blob/master/scripts/homer_mysql_rotate.pl#L236

Seeing this is a default mysql setting, I'd personally recommend adding this key size constraint here (like the others)

I'm going to make homer-docker compatible with this particular line, fwiw.

**Comment by dunst0** [comment]
@adubovikov you have pointed me to the correct solution, `SET @@global.innodb_large_prefix = 1;` fixed it but mysql 5.7 says it deprecated. Thank you very much.

**Comment by adubovikov** [comment]
glad to see that it has been fixed
